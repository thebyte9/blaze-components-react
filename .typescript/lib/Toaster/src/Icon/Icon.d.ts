/// <reference types="react" />
declare const Icon: ({ appearance, autoDismiss, autoDismissTimeout, isRunning }: any) => JSX.Element;
export default Icon;
