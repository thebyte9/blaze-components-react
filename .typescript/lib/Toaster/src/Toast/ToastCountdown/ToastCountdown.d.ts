/// <reference types="react" />
declare const ToastCountdown: ({ autoDismissTimeout, opacity, isRunning, ...props }: any) => JSX.Element;
export default ToastCountdown;
