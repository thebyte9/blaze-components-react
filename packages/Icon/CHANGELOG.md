# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.8.0-alpha.31](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.30...v0.8.0-alpha.31) (2021-09-06)


### Features

* theme Input component ([18735ec](https://github.com/thebyte9/blaze-components-react/commit/18735ec72dc2b4e49550ddf6d854b44edbe15559))







**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.30](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.29...v0.8.0-alpha.30) (2021-09-06)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.29](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.28...v0.8.0-alpha.29) (2021-07-27)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.28](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.26...v0.8.0-alpha.28) (2021-07-20)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.26](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.25...v0.8.0-alpha.26) (2021-07-20)

**Note:** Version bump only for package @blaze-react/icon







**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.25](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.24...v0.8.0-alpha.25) (2021-07-20)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.24](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.23...v0.8.0-alpha.24) (2021-07-20)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.23](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.22...v0.8.0-alpha.23) (2021-07-19)

**Note:** Version bump only for package @blaze-react/icon







**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.22](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.21...v0.8.0-alpha.22) (2021-07-19)


### Features

* component theming ([9685972](https://github.com/thebyte9/blaze-components-react/commit/9685972039dfd10f924b69b5962710c55579e27d))





# [0.8.0-alpha.21](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.20...v0.8.0-alpha.21) (2021-07-14)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.20](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.19...v0.8.0-alpha.20) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.19](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.18...v0.8.0-alpha.19) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.18](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.17...v0.8.0-alpha.18) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon







**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.17](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.16...v0.8.0-alpha.17) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.16](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.15...v0.8.0-alpha.16) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.15](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.14...v0.8.0-alpha.15) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.14](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.13...v0.8.0-alpha.14) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.13](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.12...v0.8.0-alpha.13) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.12](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.11...v0.8.0-alpha.12) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.11](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.10...v0.8.0-alpha.11) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.10](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.9...v0.8.0-alpha.10) (2021-07-13)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.9](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.8...v0.8.0-alpha.9) (2021-07-12)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.8](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.7...v0.8.0-alpha.8) (2021-07-12)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.7](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.6...v0.8.0-alpha.7) (2021-07-12)

**Note:** Version bump only for package @blaze-react/icon





# [0.8.0-alpha.6](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.3...v0.8.0-alpha.6) (2021-07-09)


### Features

* add Tab stories ([13adc96](https://github.com/thebyte9/blaze-components-react/commit/13adc969d8e58179e2ba661039b2e2c4a491e9fd))
