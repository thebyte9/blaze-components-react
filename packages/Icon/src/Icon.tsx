import React from 'react';
import { CustomIcon } from './CustomIcon';
import { IconDisplayType, IconProps } from './types';

export const Icon = ({ display, label, iconOnly, icon }: IconProps): JSX.Element => (
  <div className="flex items-center justify-center">
    {display === IconDisplayType.Left ? (
      <>
        <CustomIcon utilities="w-6 h-6 stroke-current fill-current" content={icon} />
        {!iconOnly && <span className="mx-1">{label ?? 'Blaze'}</span>}
      </>
    ) : (
      <>
        {!iconOnly && <span className="mx-1">{label ?? 'Blaze'}</span>}

        {display !== IconDisplayType.NoIcon && (
          <CustomIcon utilities="w-6 h-6 stroke-current fill-current" content={icon} />
        )}
      </>
    )}
  </div>
);
