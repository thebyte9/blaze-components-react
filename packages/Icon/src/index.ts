import CustomIcon from './CustomIcon';

export default CustomIcon;
