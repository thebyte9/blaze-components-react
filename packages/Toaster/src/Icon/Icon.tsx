import React from "react";
import { AlertIcon, CheckIcon, FlameIcon, InfoIcon } from "../Icons";
import ToastCountdown from "../Toast/ToastCountdown";

const iconsByAppearance = {
  error: FlameIcon,
  info: InfoIcon,
  success: CheckIcon,
  warning: AlertIcon
};

const Icon = ({
  appearance,
  autoDismiss,
  autoDismissTimeout,
  isRunning
}: any) => {
  const Glyph = iconsByAppearance[appearance];

  return (
    <div
      className={`react-toast-notifications__toast__icon-wrapper react-toast-notifications__toast__icon-wrapper--${appearance}`}
    >
      <ToastCountdown
        opacity={autoDismiss ? 1 : 0}
        autoDismissTimeout={autoDismissTimeout}
        isRunning={isRunning}
      />
      <Glyph className="react-toast-notifications__toast__icon" />
    </div>
  );
};
export default Icon;
