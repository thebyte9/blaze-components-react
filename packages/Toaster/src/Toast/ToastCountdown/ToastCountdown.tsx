import React from "react";

const ToastCountdown = ({
  autoDismissTimeout,
  opacity,
  isRunning,
  ...props
}: any) => (
  <div
    className="react-toast-notifications__toast__countdown_animation"
    style={{
      animationDuration: `${autoDismissTimeout}ms`,
      animationPlayState: isRunning ? "running" : "paused",
      opacity
    }}
    {...props}
  />
);

export default ToastCountdown;
