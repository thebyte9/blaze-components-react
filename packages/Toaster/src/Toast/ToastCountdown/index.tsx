import ToastCountdown from "./ToastCountdown";

export default ToastCountdown;
