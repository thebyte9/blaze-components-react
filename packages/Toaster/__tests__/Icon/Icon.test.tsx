import { cleanup, render } from "@testing-library/react";
import expect from "expect";
import React from "react";
import Icon from "../../src/Icon";

jest.mock("../../src/Toast/ToastCountdown", () =>
  jest.fn(() => <div>Toast.Countdown</div>)
);

import ToastCountdown from "../../src/Toast/ToastCountdown";

describe("Icon", () => {
  afterEach(cleanup);

  it("should be defined", () => {
    expect(Icon).toBeDefined();
  });

  it("should render withour throwing error", () => {
    const { container } = render(
      <Icon
        appearance={"success"}
        autoDismiss={true}
        autoDismissTimeout={3000}
        isRunning={false}
      ></Icon>
    );

    expect(container).toMatchSnapshot();
  });

  it("should child component recieve the expected props", () => {
    render(
      <Icon
        appearance={"success"}
        autoDismiss={true}
        autoDismissTimeout={3000}
        isRunning={false}
      ></Icon>
    );

    expect(ToastCountdown).toHaveBeenCalledWith(
      {
        autoDismissTimeout: 3000,
        isRunning: false,
        opacity: 1
      },
      {}
    );
  });
});
