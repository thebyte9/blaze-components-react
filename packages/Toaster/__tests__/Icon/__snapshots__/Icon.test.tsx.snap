// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render withour throwing error 1`] = `
<div>
  <div
    class="react-toast-notifications__toast__icon-wrapper react-toast-notifications__toast__icon-wrapper--success"
  >
    <div>
      Toast.Countdown
    </div>
    <svg
      aria-hidden="true"
      class="react-toast-notifications__toast__icon"
      fill="#FFFFFF"
      height="16"
      style="display: inline-block; vertical-align: text-top;"
      viewBox="0 0 12 16"
      width="12"
    >
      <path
        d="M12 5.5l-8 8-4-4L1.5 8 4 10.5 10.5 4 12 5.5z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
